validator/gke_restrict_client_auth_methods_test.rego

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPGKERestrictClientAuthenticationMethodsConstraintV1
+
+import data.validator.gcp.lib as lib
+
+all_violations[violation] {
+	resource := data.test.fixtures.gke_restrict_client_auth_methods.assets[_]
+	constraint := data.test.fixtures.gke_restrict_client_auth_methods.restrict_gke_client_auth_methods.constraints
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+test_restrict_client_auth_methods_violations {
```

Comment:
can we split this up into multiple tests? otherwise it's hard to tell one failure from another

Reply:
updated